### PR TITLE
Yan/lbm1 17384 - reconnecto to rabbitmq in case of failure

### PR DESCRIPTION
**Pull request by yant-lb**
**Problem**
As seen on ticket [#LBM1-17384](https://lightbitslabs.atlassian.net/browse/LBM1-17384?atlOrigin=eyJpIjoiYmI2MzVhZmI2MTg4NGM3YTg2ZDY0Y2EwNDRlNjM1MzciLCJwIjoiaiJ9) we got sometimes failures in while calling t rabbitmq.

I've implemented a reconnection that is being done on the Connection thread.
Currently it tries to reconnect once, if it fails it will continue with the original failure flow (i.e set pdb trace)

**Tests**
Iv'e created a test branch: https://github.com/LightBitsLabs/inaugurator/tree/yan/LBM1-17384-TEST which is a branch of the current branch.
Basically what it does, is throwing the ConnectionClosed exception once in the code (in order to fail the run in the first time, then in the second run it will reconnect and reproduce the original function)

The tests were done on rack03-server61
I've tested the following flows:
1. checked_in_down -> hard_reclam -> wait -> WORKS
2. checked_in_down -> hard_reclam -> wait -> allocate -> WORKS
3. checked_in_down -> hard_reclam -> allocate -> WORKS
4. checked_in_down -> allocate -> WORKS
5. hard_reclam -> allocate -> WORKS
6. hard_reclam -> WORKS

### CODE REVIEW

File: inaugurator/talktoserver.py [hunk 1]
```diff
@@ -70,6 +75,22 @@ def _connect(self, amqpURL):
...
+        logging.info("Trying to reconnect RabbitMQ.")
+        try:
+            self._cleanUpResources(set_as_finished=False)
+        except Exception as e:
+            # cleanup is not a must on reconnection we can ignore it
+            logging.exception("Couldnt cleanup resources. ignoring...")
+        try:
+            self._connect(self._amqpURL)
+        except Exception as e:
+            logging.exception("Couldnt to reconnect RabbitMQ.")
```
Comment (amittar-lb) [inline inaugurator/talktoserver.py:91]: I think it will be good to log the exception itself as well. (f"Couldnt to reconnect RabbitMQ. Reaseon: {e}") 
It can help us understand why the connection failed.

Comment (yant-lb) [reply inaugurator/talktoserver.py:91]: Thanks.
Yes, I used `logging.exception` for that: https://stackoverflow.com/a/5191885
"logger.exception will output a stack trace alongside the error message."

I think it should be enough

Comment (amittar-lb) [reply inaugurator/talktoserver.py:91]: Yes it will be good

**Comment by yant-lb** [comment]
@yuval-lb ping :+1:

**Review by amittar-lb — APPROVED** [review]
